test/test_approximate_epsilon_time_policy.cpp

```diff
@@ -0,0 +1,233 @@
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+#include <vector>
+
+
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+
+#include "message_filters/synchronizer.h"
+#include "message_filters/sync_policies/approximate_epsilon_time.h"
+#include "message_filters/message_traits.h"
+
+using namespace std::placeholders;
+using namespace message_filters;
+using namespace message_filters::sync_policies;
+
+struct Header
+{
+  rclcpp::Time stamp;
+};
+
+
+struct Msg
+{
+  Header header;
+  int data;
+};
+using MsgPtr = std::shared_ptr<Msg>;
+using MsgConstPtr = std::shared_ptr<const Msg>;
+
+namespace message_filters
+{
+namespace message_traits
+{
+template<>
+struct TimeStamp<Msg>
+{
+  static rclcpp::Time value(const Msg& m)
+  {
+    return m.header.stamp;
+  }
+};
+}
+}
+
+typedef std::pair<rclcpp::Time, rclcpp::Time> TimePair;
+typedef std::pair<rclcpp::Time, unsigned int> TimeAndTopic;
+struct TimeQuad
+{
+  TimeQuad(rclcpp::Time p, rclcpp::Time q, rclcpp::Time r, rclcpp::Time s)
+  {
+    time[0] = p;
+    time[1] = q;
+    time[2] = r;
+    time[3] = s;
+  }
+  rclcpp::Time time[4];
+};
+
+
+class ApproximateEpsilonTimeSynchronizerTest
+{
+public:
+
+  ApproximateEpsilonTimeSynchronizerTest(const std::vector<TimeAndTopic> &input,
+				  const std::vector<TimePair> &output,
+				  uint32_t queue_size, rclcpp::Duration epsilon) :
+    input_(input), output_(output), output_position_(0), sync_(
+      ApproximateEpsilonTime<Msg, Msg>{queue_size, epsilon})
+  {
+    sync_.registerCallback(std::bind(&ApproximateEpsilonTimeSynchronizerTest::callback, this, _1, _2));
+  }
+
+  void callback(const MsgConstPtr& p, const MsgConstPtr& q)
+  {
+    ASSERT_TRUE(p);
+    ASSERT_TRUE(q);
+    ASSERT_LT(output_position_, output_.size());
+    EXPECT_EQ(output_[output_position_].first, p->header.stamp);
+    EXPECT_EQ(output_[output_position_].second, q->header.stamp);
+    ++output_position_;
+  }
+
+  void run()
+  {
+    for (unsigned int i = 0; i < input_.size(); i++)
+    {
+      if (input_[i].second == 0)
+      {
+        MsgPtr p(std::make_shared<Msg>());
+        p->header.stamp = input_[i].first;
+        sync_.add<0>(p);
+      }
+      else
+      {
+        MsgPtr q(std::make_shared<Msg>());
+        q->header.stamp = input_[i].first;
+        sync_.add<1>(q);
+      }
+    }
+    //printf("Done running test\n");
```

Comment:
```suggestion
```

Reply:
Removed in https://github.com/ros2/message_filters/pull/84/commits/afd1542fe23807fec355fb6c5bc8563d0f347854